src/lib/radio/radio.ts

```diff
@@ -180,14 +185,13 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
     let isAlreadySelected = this._selected != null && this._selected.value == this._value;
 
     if (this._radios != null && !isAlreadySelected) {
-      let matchingRadio = this._radios.filter(radio => radio.value == this._value)[0];
-
-      if (matchingRadio) {
-        this.selected = matchingRadio;
-      } else if (this.value == null) {
-        this.selected = null;
-        this._radios.forEach(radio => { radio.checked = false; });
-      }
+      this._selected = null;
+      this._radios.forEach(radio => {
+        radio.checked = this.value == radio.value;
+        if (radio.checked) {
+          this._selected = radio;
+        }
```

Comment:
The `selected()` does 3 things: update `_selected` (which is done); update `value` (`_updateSelectedRadioFromValue` is always called from value change) ; check `selected` radio button (we get the `_selected` value from checked radio buttons, and `checked` is set on line 190).